src/linkPanel/widgets/linking.ts

```diff
@@ -161,48 +218,41 @@ export class Linking extends LinkEditorWidget {
     panel.title.label = 'Advanced linking';
 
     const glueToolbar = new Toolbar();
-    const attributes = new BoxPanel({ direction: 'left-to-right' });
 
     const advancedSelect = new AdvancedLinkingChoices({
-      categories: this._linkEditorModel.availableAdvancedLinks
-    });
-
-    this._advancedToolbar.push({
-      name: 'Select advanced',
-      widget: ReactWidget.create(advancedSelect.render())
+      categories: this._linkEditorModel.advLinksPromise
     });
+    advancedSelect.onChange.connect(this.onAdvancedLinkChanged, this);
+    glueToolbar.addItem(
+      'Select advanced',
+      ReactWidget.create(advancedSelect.render())
+    );
```

Comment:
Generally speaking, this is an anti-pattern way of using React, the parent component does not need to listen for the changed event of its children, the child itself needs to report its changes to the parent through the `props`. For example:

```js
  <AdvancedLinkingChoices
    categories={this._linkEditorModel.advLinksPromise}
    onAdvancedLinkChanged={this.onAdvancedLinkChanged}
  />
```
Then inside the `AdvancedLinkingChoices` component, it will call the `props.onAdvancedLinkChanged` function on its changed event.

In this use case of `AdvancedLinkingChoices`, I would prefer to stick with a normal lumino widget instead of using React.